blockdevice/probe/probe.go

```diff
@@ -34,106 +26,96 @@ type ProbedBlockDevice struct {
 	Path       string
 }
 
-// All probes a block device's file system for the given label.
-func All() (all []*ProbedBlockDevice, err error) {
-	var infos []os.FileInfo
-
-	if infos, err = ioutil.ReadDir("/sys/block"); err != nil {
-		return nil, err
-	}
-
-	for _, info := range infos {
-		devpath := "/dev/" + info.Name()
+// SelectOption is a callback matcher for All block devices probes.
+type SelectOption func(device *ProbedBlockDevice) (bool, error)
 
-		var probed []*ProbedBlockDevice
-
-		probed, err = probeFilesystem(devpath)
+// WithPartitionsLabeled search for a block device which has partitions with some specific labels.
+func WithPartitionsLabeled(label string) SelectOption {
```

Comment:
```suggestion
func WithPartitionLabel(label string) SelectOption {
```

nit, I think this sounds better